zkstack_cli/crates/zkstack/src/commands/autocomplete.rs

```diff
@@ -0,0 +1,57 @@
+use std::{
+    fs::File,
+    io::{BufWriter, Write},
+};
+
+use anyhow::Context;
+use clap::CommandFactory;
+use clap_complete::{generate, Generator};
+use common::logger;
+
+use super::args::AutocompleteArgs;
+use crate::{
+    messages::{msg_generate_autocomplete_file, MSG_OUTRO_AUTOCOMPLETE_GENERATION},
+    ZkStack,
+};
+
+pub fn run(args: AutocompleteArgs) -> anyhow::Result<()> {
+    let path = args.out.join(args.generator.autocomplete_file_name());
+
+    logger::info(msg_generate_autocomplete_file(
+        path.to_str()
+            .context("the output file path is an invalid UTF8 string")?,
+    ));
+
+    let file = File::create(path).context("Failed to create file")?;
+    let mut writer = BufWriter::new(file);
+
+    generate_completions(args.generator, &mut writer)?;
+
+    logger::outro(MSG_OUTRO_AUTOCOMPLETE_GENERATION);
+
+    Ok(())
+}
+
+pub fn generate_completions<G: Generator>(gen: G, buf: &mut dyn Write) -> anyhow::Result<()> {
+    let mut cmd = ZkStack::command();
+    let cmd_name = cmd.get_name().to_string();
+
+    generate(gen, &mut cmd, cmd_name, buf);
+
+    Ok(())
+}
+
+pub trait ShellAutocomplete {
+    fn autocomplete_file_name(&self) -> &'static str;
+}
+
+impl ShellAutocomplete for clap_complete::Shell {
```

Comment:
Looks like unnecessary complexity

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3097/commits/0fee9d53e736e2147a4b4c991d7b59bbd64c6dba